pkg/sidecar/docker_reactor.go

```diff
@@ -174,6 +174,25 @@ func (d *DockerReactor) handleContainer(ctx context.Context, container *docker.C
 		controlRoutes = append(controlRoutes, nlroutes...)
 	}
 
+	// Allow traffic from the host machine (acting as the gateway).
+	var (
+		gwIPs    = make(map[string]struct{})
+		gwRoutes []netlink.Route
+	)
+	for _, route := range controlRoutes {
+		if _, ok := gwIPs[route.Gw.String()]; ok {
+			continue
+		}
+		nlroutes, err := netlinkHandle.RouteGet(route.Gw)
+		if err != nil {
+			return nil, fmt.Errorf("failed to resolve route: %w", err)
+		}
+		gwIPs[route.Gw.String()] = struct{}{}
+		gwRoutes = append(gwRoutes, nlroutes...)
+	}
+
+	controlRoutes = append(controlRoutes, gwRoutes...)
```

Comment:
This is the key. In kubernetes we don't need this, because kubernetes has native port-forwarding out of the box. In our case, we connect our test containers to the control and data networks. The former is tightly restricted, and the latter is internal and connects only the containers participating in the test. For containers to successfully expose ports to the host, besides publishing those ports, we need to ALLOW traffic from the gateway (host).

Reply:
Just for comparison... Without this change, from the current master, 
the route table looks like this:
```
docker exec tg-example-prometheus2-19979292b320-single-1 ip route
16.1.0.0/16 dev eth1 proto kernel scope link src 16.1.0.3 
192.18.0.3 dev eth0 src 192.18.0.7 
192.18.0.4 dev eth0 src 192.18.0.7 
```
and there is no *default* route, so no place to send packets outside of known networks.
```
docker exec tg-example-prometheus2-19979292b320-single-1 ping -c 2 google.com
ping: google.com: Temporary failure in name resolution
```

Reply:
Thanks for spotting this. Definitely unintended. I've fixed this logic in eaa06cd. Can you take it out for a spin again? Works for me.

* The host can access ports exposed by the container.
* The container cannot access public IPs.